frontends/mrxl/mrxl/gen_calyx.py

```diff
@@ -296,8 +217,17 @@ def gen_stmt_impl(
 
     name2par maps memory names to banking factors.
     """
-    if isinstance(stmt.operation, ast.Map):
-        return gen_map_impl(
+    if isinstance(stmt.operation, ast.Map) and use_my_map_impl:
+        return my_map_impl(
+            comp,
+            stmt.dst,
+            stmt.operation,
+            arr_size,
+            name2par[stmt.dst],
+            statement_idx,
+        )
+    elif isinstance(stmt.operation, ast.Map):
+        return map_impl.gen_map_impl(
```

Comment:
This could instead look like:

```py
if isinstance(stmt.operation, ast.Map):
    gen_map = my_map_impl if use_my_map_impl else map_impl.gen_map_impl
    return gen_map(
        ...
    )
```

to avoid duplicating the arguments.